Mathlib/RingTheory/IntegralClosure/IsIntegral/Basic.lean

```diff
@@ -214,3 +214,22 @@ theorem isNoetherian_adjoin_finset [IsNoetherianRing R] (s : Finset A)
   isNoetherian_of_fg_of_noetherian _ (fg_adjoin_of_finite s.finite_toSet hs)
 
 end
+
+section Prod
+
+variable {R A B : Type*}
+variable [CommRing R] [Ring A] [Ring B] [Algebra R A] [Algebra R B]
+
+/-- An element of a product algebra is integral if each component is integral. -/
+theorem IsIntegral.pair {a : A × B} (ha₁ : IsIntegral R a.1) (ha₂ : IsIntegral R a.2) :
+    IsIntegral R a := by
+  obtain ⟨μ₁, ⟨hμ₁Monic, hμ₁Eval⟩⟩ := ha₁
+  obtain ⟨μ₂, ⟨hμ₂Monic, hμ₂Eval⟩⟩ := ha₂
+  refine ⟨μ₁ * μ₂, ⟨hμ₁Monic.mul hμ₂Monic, ?_⟩⟩
+  rw [← Polynomial.aeval_def] at *
+  rw [Polynomial.aeval_prod_apply,
+    Polynomial.aeval_mul, hμ₁Eval, zero_mul,
+    Polynomial.aeval_mul, hμ₂Eval, mul_zero]
+  rfl
```

Comment:
The Polynomial namespace is already open.
μ is a weird name for a polynomial. Why not p?
```suggestion
  obtain ⟨μ₁, hμ₁Monic, hμ₁Eval⟩ := ha₁
  obtain ⟨μ₂, hμ₂Monic, hμ₂Eval⟩ := ha₂
  refine ⟨μ₁ * μ₂, hμ₁Monic.mul hμ₂Monic, ?_⟩
  rw [← aeval_def] at *
  rw [aeval_prod_apply, aeval_mul, hμ₁Eval, zero_mul, aeval_mul, hμ₂Eval, mul_zero, Prod.zero_eq_mk]
```